source/tests/system/nirfsg_driver_api_tests.cpp

```diff
@@ -0,0 +1,130 @@
+#include <gtest/gtest.h>
+
+#include "device_server.h"
+#include "nirfsg/nirfsg_client.h"
+#include "nirfsg/nirfsg_service.h"
+
+namespace ni {
+namespace tests {
+namespace system {
+
+namespace rfsg = nirfsg_grpc;
+namespace client = nirfsg_grpc::experimental::client;
+using namespace ::testing;
+
+const int krfsgDriverApiSuccess = 0;
+
+class NiRFSGDriverApiTests : public ::testing::Test {
+ protected:
+  NiRFSGDriverApiTests()
+      : device_server_(DeviceServerInterface::Singleton()),
+        nirfsg_stub_(rfsg::NiRFSG::NewStub(device_server_->InProcessChannel()))
+  {
+    device_server_->ResetServer();
+  }
+  virtual ~NiRFSGDriverApiTests() {}
+
+  void SetUp() override
+  {
+    initialize_driver_session();
+  }
+
+  void TearDown() override
+  {
+    close_driver_session();
+  }
+
+  std::unique_ptr<rfsg::NiRFSG::Stub>& GetStub()
+  {
+    return nirfsg_stub_;
+  }
+
+  int GetSessionId()
+  {
+    return driver_session_->id();
+  }
+
+  void initialize_driver_session()
+  {
+    const auto DEVICE = "FakeDevice";
+    const auto OPTIONS = "Simulate=1, DriverSetup=Model:5652; BoardType:PXI";
+    auto response = client::init_with_options(
+        GetStub(),
+        DEVICE,
+        false,
+        false,
+        OPTIONS);
+
+    ASSERT_EQ(krfsgDriverApiSuccess, response.status());
+    driver_session_ = std::make_unique<nidevice_grpc::Session>(response.vi());
+  }
+
+  void close_driver_session()
+  {
+    ::grpc::ClientContext context;
```

Comment:
I guess you added these before our previous thread? You shouldn't need these anymore. I'd also recommend a more concise name for `GetSessionId` so that you can inline client calls (at least most of the time): `auto response = client::whatever(stub(), session(), ...)`

Reply:
Yeah, you're right. I cleaned this all up, thanks!